### PR TITLE
[API change] Added convenient upload method to BlobClient

### PR DESCRIPTION
**Pull request by gapra-msft**
Resolves https://github.com/Azure/azure-sdk-for-java/issues/6089

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java [hunk 1]
```diff
@@ -115,6 +120,70 @@ public PageBlobClient getPageBlobClient() {
...
+        final ParallelTransferOptions validatedParallelTransferOptions =
+            ModelHelper.populateAndApplyDefaults(parallelTransferOptions);
+        if (length < validatedParallelTransferOptions.getMaxSingleUploadSize()) {
+            blockBlobClient.uploadWithResponse(data, length, headers, metadata, tier, null, requestConditions,
+                timeout, context);
+        } else {
+            BlobOutputStream blobOutputStream = BlobOutputStream.blockBlobOutputStream(client,
+                validatedParallelTransferOptions, headers, metadata, tier, requestConditions);
+            try {
+                data.transferTo(blobOutputStream);
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java]: I thought this method was only java 9?

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java]: yep, I need to change it

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClient.java]: I made the PR before I realized

File: sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java [hunk 2]
```diff
@@ -241,4 +244,27 @@ public static OffsetDateTime parseDate(String dateString) {
...
     }
+
+    /**
+     * Reads data from an input stream and writes it to an output stream.
+     * @param source {@link InputStream source}
+     * @param writeLength The length of data to write.
+     * @param destination {@link OutputStream destination}
+     * @throws IOException If an I/O error occurs.
+     */
+    public static void copyToOutputStream(InputStream source, long writeLength, OutputStream destination)
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java]: Could this go into `StorageImplUtils` for the time being? Do we expected or would want a customer to use this API in the near-term?

Comment (gapra-msft) [reply sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java]: I have nothing against doing that

Comment (gapra-msft) [reply sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/Utility.java]: done

**Pull request merged**